### DISCUSSION TITLE
Why was `name` removed from Binary?

### DISCUSSION THREAD

**Discussion by fzakaria** [lief-project/LIEF#963] [General]
I noticed as part of https://github.com/lief-project/LIEF/issues/839 `name()` for the Binary object class was removed.
I can't tell the rationale from the PR why?

It's not a big deal--- I can change my code to keep pairs of `(str, lief.Binary)` but I'm more curious as to why the change was made.

**Comment by romainthomas** [comment]
The reason why `name` has been removed in that it is only meaningful when parsing a binary from a file path.
LIEF exposes other parsing interface through *stream* where this information is not available. 

The second reason is the current trend to try to keep the sizeof LIEF objects as small as possible and the binary's name can be managed by the user as you plan to do pretty easily.

**Reply by fzakaria** [reply]
Here is what I ended up doing if others want to get back to a similar experience:
```python
# pyright: strict
from typing import TYPE_CHECKING, Any

import lief

# Let's make sure type checking works for this proxy class
# https://stackoverflow.com/questions/71365594/how-to-make-a-proxy-object-with-typing-as-underlying-object-in-python
if TYPE_CHECKING:
    base = lief.ELF.Binary
else:
    base = object


class Binary(base):
    """Proxy the lief.Binary object to add a path attribute.

    As of https://github.com/lief-project/LIEF/issues/839 the name
    attribute in lief.Binary was removed. Rather than passing around
    a tuple let's create a nice proxy class.
    """

    def __init__(self, path: str):
        self.path = path
        self.__binary = lief.parse(path)

    if not TYPE_CHECKING:

        def __getattr__(self, attr: str) -> Any:
            return getattr(self.__binary, attr)

    @staticmethod
    def is_elf(path: str):
        return lief.is_elf(path)
```
